src/Drupal/Driver/Cores/Drupal7.php

```diff
@@ -482,4 +482,18 @@ public function clearStaticCaches() {
     drupal_static_reset();
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function entityCreate($entity_type, $entity) {
+    // @todo: create a D7 version of this function
```

Comment:
And here too. Alternatively, since D7 *does* support entities, throw the exception here and note it is a todo item.